pytorch_lightning/trainer/training_loop.py

```diff
@@ -870,7 +870,10 @@ def run_training_batch(self, batch, batch_idx):
                 self.batch_loss_value.append(opt_closure_result.loss)
 
                 # track all the outputs across all steps
-                batch_outputs[opt_idx].append(opt_closure_result.training_step_output_for_epoch_end)
+                if len(batch_outputs) == 1:
+                    batch_outputs[0].append(opt_closure_result.training_step_output_for_epoch_end)
+                else:
+                    batch_outputs[opt_idx].append(opt_closure_result.training_step_output_for_epoch_end)
```

Comment:
```suggestion
                idx = opt_idx if len(batch_outputs) > 1 else 0
                batch_outputs[idx].append(opt_closure_result.training_step_output_for_epoch_end)
```
maybe find better name for "idx"